matrix-sdk/src/main/java/org/matrix/androidsdk/data/timeline/MXEventTimeline.java

```diff
@@ -470,10 +471,10 @@ private void addPaginationEvents(List<Event> events,
             for (Event stateEvent : stateEvents) {
                 if (direction == Direction.BACKWARDS) {
                     // Enrich the timeline root state with the additional state events observed during back pagination
-                    processStateEvent(stateEvent, Direction.FORWARDS);
+                    processStateEvent(stateEvent, Direction.FORWARDS, true);
                 }
 
-                processStateEvent(stateEvent, direction);
+                processStateEvent(stateEvent, Direction.BACKWARDS, true);
```

Comment:
There is an issue here when direction is Forward, we never call processStateEvent

Reply:
Yes, thx